src/osd/OSDMap.cc

```diff
@@ -279,6 +279,8 @@ int OSDMap::Incremental::propagate_snaps_to_tiers(CephContext *cct,
 	if (new_rem_it != new_removed_snaps.end()) {
 	  new_removed_snaps[tier_pool] = new_rem_it->second;
 	}
+
+	tier->application_metadata = base.application_metadata;
```

Comment:
instead of overwriting the application_metadata on tier, shall we instead `insert_or_assign()` all elements from base to tier?

Reply:
I'm thinking no, because the mon doesn't allow you to set metadata on a cache tier in the first place.  

I think the only situation this could break is if there were a really old cluster that somehow ended up with metadata set that way on the cache and not the base, but the tier check was introduced in b0e7b0c55393337a9b8caed7aa7767e0e3f3f190 and application_metadata in a15f6e4cea5af9fc82c755120642634b68af5992, about 10 commits apart in the luminous dev cycle.